packages/blocks/src/api/factory.js

```diff
@@ -163,14 +153,14 @@ export function cloneBlock( block, mergeAttributes = {}, newInnerBlocks ) {
  * @return {boolean} Is the transform possible?
  */
 const isPossibleTransformForSource = ( transform, direction, blocks ) => {
-	if ( isEmpty( blocks ) ) {
+	if ( ! blocks.length ) {
```

Comment:
Adding a note that blocks could in theory be `undefined` (in which case `blocks?.length` would be preferred), but the rest of the code on this file already appears to assume that it will always be an array 👍

Reply:
Tests pass so it’s probably working correctly but extra safety won’t hurt. I believe that `isEmpty` like most lodash methods was covering a few things including checking if we deal with an array in the first place.

Reply:
Yeah, this is a common problem with migrating away from Lodash, especially in untyped codebases 😕 It's hard to separate which checks we want to keep, and which ones are not needed. Implicit conversions are even worse.

Sometimes you get it wrong and create some bugs that need to be fixed, but the end result after all that is much better: not only are you not relying on a large lib anymore, but you're now explicit about your checks, which improves readability and maintainability. It's painful, but it's worth it 🙂

Reply:
TBH cases like `isEmpty` in combination with the untyped codebase makes it the ultimate hell for following through code and ensuring no bugs occur. By removing the extra ambiguity that `isEmpty()` added, we solve one of the problems, and then when one day we type the entire codebase, things will look great!